backend/src/services/update-user/update-user.service.ts

```diff
@@ -0,0 +1,58 @@
+import { prisma } from '@data/prisma-client';
+import { bcryptHash } from '@helpers/crypto/crypto';
+import { Prisma, User } from '@prisma/client';
+
+const updateUser = async (
+  user: Prisma.UserCreateInput,
+  new_password?: string,
+): Promise<User> => {
+  const { id, ...userData } = user;
+  console.log(userData);
+
+  if (new_password) {
+    await updatePassword(new_password, id);
+  }
+
+  const updatedUser = await prisma.user.update({
+    where: {
+      id,
+    },
+    data: {
+      ...userData,
+    },
+  });
+
+  return updatedUser;
+};
+
+const updatePassword = async (
+  newPassword: string,
+  id?: string,
+): Promise<void> => {
```

Comment:
Generally, it's better to order arguments from less specific to more specific. So I would pass the `id` of the user as the first arg